File: Part 2 - Regression/Section 9 - Random Forest Regression/random_forest_regression.R
# Random Forest Regression

# Importing the dataset
dataset <-  read.csv('Position_Salaries.csv')
dataset <- dataset[2:3]

#Dont need to split into training and test set since the dataset is small
#Dont need feature scaling since the decision trees dont depend on Euclidean distances of the 
#data points

# Fitting Random Forest Regression to the dataset
#install.packages('randomForest')
library(randomForest)
set.seed(1234)
regressor <- randomForest(x = dataset_model[1], #we need a dataframe hence brackets 
                          y = dataset_model$Salary, # dollar sign because this needs vector
                          ntree = 450)

summary(regressor)
# Predicting a new result with Random Forest Regression
y_pred <- predict(regressor, data.frame(Level = 6.5))

# Visualising the Random Forest Regression results (higher resolution)
# install.packages('ggplot2')
library(ggplot2)
x_grid = seq(min(dataset$Level), max(dataset$Level), 0.01)
ggplot() +
  geom_point(aes(x = dataset$Level, y = dataset$Salary),
             colour = 'red') +
  geom_line(aes(x = x_grid, y = predict(regressor, newdata = data.frame(Level = x_grid))),
            colour = 'blue') +
  ggtitle('Truth or Bluff (Random Forest Regression)') +
  xlab('Level') +
  ylab('Salary')

File: Part 2 - Regression/Section 9 - Random Forest Regression/random_forest_regression.py
# -*- coding: utf-8 -*-
"""
Created on Mon Aug 27 00:22:08 2018

@author: Hrishika
"""

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

# Importing the dataset
dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:, 1:2].values
y = dataset.iloc[:, 2].values

#Not splitting the dataset into training and test because the dataset is too small
#Feature scaling is not required since this model does not depend on Euclidean distances

#Build the regressor
from sklearn.ensemble import RandomForestRegressor
 regressor = RandomForestRegressor(n_estimators= 300, random_state= 0)
regressor.fit(X,y)


# Predicting a new result
y_pred = regressor.predict(6.5)

# Visualising the Random Forest Regression results (higher resolution)
X_grid = np.arange(min(X), max(X), 0.01)
X_grid = X_grid.reshape((len(X_grid), 1))
plt.scatter(X, y, color = 'red')
plt.plot(X_grid, regressor.predict(X_grid), color = 'blue')
plt.title('Truth or Bluff (Random Forest Regression)')
plt.xlabel('Position level')
plt.ylabel('Salary')
plt.show()